packages/hono/src/mod.test.ts

```diff
@@ -0,0 +1,86 @@
+import { assertEquals } from "@std/assert";
+import { type ContextDataFactory, federation } from "./mod.ts";
+
+interface MockHonoContext {
+  req: {
+    raw: Request;
+  };
+  res: Response;
+}
+
+interface MockFederation<T> {
+  fetch(request: Request, options: any): Promise<Response>;
+}
+
+Deno.test("federation middleware", async (t) => {
```

Comment:
In order to make this test runnable on Node.js and Bun, we should use [`node:test`](https://nodejs.org/api/test.html) and [`node:assert/strict`](https://nodejs.org/api/assert.html).